From e97ffc1f820beff12bb8084e6337168a1cd27540 Mon Sep 17 00:00:00 2001
From: Liu Yiding <liuyd.fnst@fujitsu.com>
Date: Sat, 20 Sep 2025 06:50:17 +0000
Subject: [PATCH] Fix Service start error

change "fips=no" to "-fips"
based on discussions with the OpenSSL developers in
https://github.com/FreeRADIUS/freeradius-server/issues/5631

Upstream-Status: Backport
https://github.com/FreeRADIUS/freeradius-server/commit/59e262f1134fef8d53d15ae963885a08c9ea8315

Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>
---
 src/main/tls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/tls.c b/src/main/tls.c
index 2a348eb9bb..02a4c24f70 100644
--- a/src/main/tls.c
+++ b/src/main/tls.c
@@ -3644,7 +3644,7 @@ int tls_global_init(TLS_UNUSED bool spawn_flag, TLS_UNUSED bool check)
 	CONF_modules_load_file(NULL, NULL, 0);
 
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
-	EVP_set_default_properties(NULL, "fips=no");
+	EVP_set_default_properties(NULL, "-fips");
 #endif
 
 	/*
-- 
2.43.0

